pkg/controller/nodemaintenance/check_lease_update.go

```diff
@@ -0,0 +1,431 @@
+				&coordv1beta1.Lease{
+					ObjectMeta :metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							metav1.OwnerReference{
+								APIVersion: "v1",
+								Kind:       "Node",
+								Name:       "@",
+								UID:        "#",
+							},
+						},
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString("miau"),
+						LeaseDurationSeconds: makeInt32(44),
+						AcquireTime:          getTimeLease(42),
+						RenewTime:            getTimeLease(43),
+						LeaseTransitions:     makeInt32(3),
+					},
+				},
+				&coordv1beta1.Lease{
+					ObjectMeta :metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{ {
+								APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
+								Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
+								Name:       getMockNode().Name,
+								UID:        getMockNode().UID,
+						} },
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          getTimeLease(int64(NowTime)),
+						RenewTime:            getTimeLease(int64(NowTime)),
+						LeaseTransitions:     makeInt32(4),
+					},
+				},
+			 ),
+
+		Entry("extend lease if same holder and zero renew time (invalid lease)",
+				&coordv1beta1.Lease{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							metav1.OwnerReference{
+								APIVersion: "v1",
+								Kind:       "Node",
+								Name:       "@",
+								UID:        "#",
+							},
+						},
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: nil,
+						AcquireTime:          getTimeLease(int64(NowTime-599)),
+						RenewTime:            nil,
+						LeaseTransitions:     makeInt32(3),
+						},
+				},
+				&coordv1beta1.Lease{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{ {
+								APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
+								Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
+								Name:       getMockNode().Name,
+								UID:        getMockNode().UID,
+						} },
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          getTimeLease(int64(NowTime)),
+						RenewTime:            getTimeLease(int64(NowTime)),
+						LeaseTransitions:     makeInt32(4),
+					},
+				},
+			 ),
+
+		Entry("extend lease if same holder and expired lease - check modified lease duration",
```

Comment:
I would like this test to be:

```suggestion
		Entry("extend lease if same holder and expired lease - check modified lease duration",
```

To test the future scenario where the admin modifies the expected maintenance window.

Reply:
changed test title.